### PR TITLE
Working fft

### PR DESCRIPTION
**Pull request by xSojka**
todo:
correct mkl lib path

### CODE REVIEW

**Comment by AaronWantoch** [comment]
I worked on this branch so I added my changes to it (reading from wav file)
here is a screen of a chart made from first 1024 read data
image

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 3.21)
 
 project(GuitarTrainer)
 
+include_directories(/usr/include/mkl/)
```
Comment (bstrzelecki) [inline CMakeLists.txt]: This will break in non debian distribution

File: test/e2eTest.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    Segment seg(probeSize);
+    FFT fft(probeSize);
+    size_t count = 0;
+    while(!wav_hdr.FinishedReading())
+    {
+        wav_hdr.ReadRawData(&seg);
+        fft.Calculate(&seg);
+        for(int i=0; i < probeSize; i++)
+        {
+            std::cout<<count<<" " << seg.rawSignal[i] <<" "<<seg.afterFFT[i]<<std::endl;
```
Comment (bstrzelecki) [inline test/e2eTest.cpp]: Use assertions instead of stdout

File: src/WavHandler.cpp [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        cout << "Block align                :" << wavHeader.blockAlign << endl;
+        cout << "Data string                :" << wavHeader.Subchunk2ID[0] << wavHeader.Subchunk2ID[1] << wavHeader.Subchunk2ID[2] << wavHeader.Subchunk2ID[3] << endl;
+    }
+}
+
+void WavHandler::ReadRawData(Segment *seg)
+{
+    size_t bytesRead;
+    const uint16_t BUFFER_SIZE = sizeof(seg->rawSignal[0]);
+    int8_t* buffer = new int8_t[BUFFER_SIZE];
```
Comment (bstrzelecki) [inline src/WavHandler.cpp]: Do we need to reallocate buffer every time, since BUFFER_SIZE is constant?

**Review by bstrzelecki — CHANGES_REQUESTED** [review]
add a.out to .gitignore

File: test/e2eTest.cpp (hunk 2 above)
Comment (AaronWantoch) [reply test/e2eTest.cpp]: I just wanted to have a test where we could test all modules. It is hard for me to do assertion here as it is just data from file which I have no other way of reading.

File: src/WavHandler.cpp (hunk 3 above)
Comment (AaronWantoch) [reply src/WavHandler.cpp]: no changing it

File: CMakeLists.txt (hunk 1 above)
Comment (AaronWantoch) [reply CMakeLists.txt]: @Michanowski9 your part

**Review by bstrzelecki — APPROVED** [review]

**Review by BognaLew — APPROVED** [review]

**Pull request merged**